ahcore/callbacks/abstract_writer_callback.py

```diff
@@ -333,6 +367,31 @@ def _monitor_and_cleanup(self) -> None:
             time.sleep(0.5)  # Short sleep to yield control and reduce CPU usage
 
     def _epoch_end(self, trainer: pl.Trainer, pl_module: pl.LightningModule) -> None:
+        for callback in self._callbacks:
+            while self._num_filenames_seen != callback.completed_tasks:
+                logger.debug(
```

Comment:
I tried training a segmentation model using these callbacks. In the manifest file of the training dataset, the filename was set to the absolute path (check [here](https://github.com/NKI-AI/ahcore/blob/046cd1f68bbb31df8ef4edaae67add4dfcbd3950/tools/populate_tcga_db.py#L105)). Whereas, it should be actually be the path relative to the data directory. But in any case, using a faulty manifest makes for a good test case.

When I use the faulty manifest such as above, I expect that this callback breaks the main process and throws an error. But right now, even though the worker process throws an error in the logs, it doesn't break the main process. Whereas, the main process waits indefinitely for the workers to return some result. Check the two screenshots below. 
![image](https://github.com/NKI-AI/ahcore/assets/26798611/182a8c60-74b4-44a7-8c7a-96cfe33446fb)
![image](https://github.com/NKI-AI/ahcore/assets/26798611/f8b449ac-3c68-44a4-8648-dd83a40f0594)